src/XCCDF/result.c

```diff
@@ -1067,16 +1080,32 @@ xmlNode *xccdf_target_identifier_to_dom(const struct xccdf_target_identifier *ti
 	}
 }
 
-xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info)
+xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info, struct xccdf_benchmark *benchmark, bool use_stig_rule_id)
 {
+	/* Handle attributes */
+	const char *idref = xccdf_rule_result_get_idref(result);
+	if (benchmark && use_stig_rule_id) {
+		const char *stig_rule_id = NULL;
+		struct xccdf_item *item = xccdf_benchmark_get_member(benchmark, XCCDF_RULE, idref);
+		struct oscap_reference_iterator *references = xccdf_item_get_references(XRULE(item));
+		while (oscap_reference_iterator_has_more(references)) {
+			struct oscap_reference *ref = oscap_reference_iterator_next(references);
+			if (strcmp(ref->href, DISA_STIG_VIEWER_HREF) == 0)
+				stig_rule_id = ref->title;
+		}
+		oscap_reference_iterator_free(references);
+
+		if (stig_rule_id)
+			idref = stig_rule_id;
+		else
+			return;
```

Comment:
code style, plus this definitely needs a comment. It looks like we are avoiding export of rules that don't have a stig viewer ID, this should be explained here IMO